consensus/src/threshold_simplex/actors/resolver/actor.rs

```diff
@@ -99,68 +101,59 @@ impl Inflight {
 
 /// Requests are made concurrently to multiple peers.
 pub struct Actor<
-    E: Clock + GClock + Rng + Metrics + Spawner,
-    C: PublicKey,
-    B: Blocker<PublicKey = C>,
-    V: Variant,
+    E: Clock + GClock + Rng + CryptoRng + Metrics + Spawner,
+    P: PublicKey,
+    S: SigningScheme,
+    B: Blocker<PublicKey = P>,
     D: Digest,
-    S: ThresholdSupervisor<
-        Index = View,
-        PublicKey = C,
-        Identity = V::Public,
-        Polynomial = Vec<V::Public>,
-    >,
 > {
     context: ContextCell<E>,
+    signing: S,
+
     blocker: B,
-    supervisor: S,
 
     epoch: Epoch,
     namespace: Vec<u8>,
 
-    notarizations: BTreeMap<View, Notarization<V, D>>,
-    nullifications: BTreeMap<View, Nullification<V>>,
+    notarizations: BTreeMap<View, Notarization<S, D>>,
+    nullifications: BTreeMap<View, Nullification<S>>,
     activity_timeout: u64,
 
     required: BTreeSet<Entry>,
     inflight: Inflight,
     retry: Option<SystemTime>,
 
-    mailbox_receiver: mpsc::Receiver<Message<V, D>>,
+    mailbox_receiver: mpsc::Receiver<Message<S, D>>,
 
     fetch_timeout: Duration,
     max_fetch_count: usize,
     fetch_concurrent: usize,
-    requester: requester::Requester<E, C>,
+    requester: requester::Requester<E, P>,
 
     unfulfilled: Gauge,
     outstanding: Gauge,
     served: Counter,
 }
 
 impl<
-        E: Clock + GClock + Rng + Metrics + Spawner,
-        C: PublicKey,
-        B: Blocker<PublicKey = C>,
-        V: Variant,
+        E: Clock + GClock + Rng + CryptoRng + Metrics + Spawner,
+        P: PublicKey,
+        S: SigningScheme,
+        B: Blocker<PublicKey = P>,
         D: Digest,
-        S: ThresholdSupervisor<
-            Index = View,
-            PublicKey = C,
-            Identity = V::Public,
-            Polynomial = Vec<V::Public>,
-        >,
-    > Actor<E, C, B, V, D, S>
+    > Actor<E, P, S, B, D>
 {
-    pub fn new(context: E, cfg: Config<C, B, S>) -> (Self, Mailbox<V, D>) {
+    pub fn new(context: E, cfg: Config<P, S, B>) -> (Self, Mailbox<S, D>) {
         // Initialize requester
         let config = requester::Config {
             public_key: cfg.crypto,
             rate_limit: cfg.fetch_rate_per_peer,
             initial: cfg.fetch_timeout / 2,
             timeout: cfg.fetch_timeout,
         };
-        let requester = requester::Requester::new(context.with_label("requester"), config);
+        let mut requester = requester::Requester::new(context.with_label("requester"), config);
+        let participants: Set<_> = cfg.participants.into_iter().collect();
+        requester.reconcile(participants.as_ref());
```

Comment:
Since the participants set is now stable throughout the engine lifetime, we only need to reconcile once on startup.